### PR TITLE
Add Python 3 support.

### PR DESCRIPTION
**Pull request by sunu**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 MINIMUM_CYTHON_VERSION = '0.13'
 
+
+def cmp(a, b):
```
Comment (axiak) [inline setup.py]: where is this used?

Comment (sunu) [inline setup.py]: @axiak  At https://github.com/axiak/pyre2/blob/master/setup.py#L26. `cmp` function is not there in Python 3; so this is to replace it.

Comment (axiak) [inline setup.py]: ah, great.

**Comment by axiak** [comment]
LGTM

**Comment by sunu** [comment]
Hey @axiak !

Just to make sure, was I supposed to include the generated changes to re.cpp as well or is it done by you during a release? 

Thanks.

**Pull request merged**